app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -7,8 +7,7 @@ def create_resource(_type, _id = nil, data = {})
 
       klass = CloudVolume.class_by_ems(ext_management_system)
 
-      validate = klass.validate_create_volume(ext_management_system)
-      raise validate[:message] unless validate[:available]
+      raise BadRequestError, ext_management_system.unsupported_reason(:cloud_volume_create) unless ext_management_system.supports?(:cloud_volume_create)
```

Comment:
@agrare @Fryguy Lines 54-61 of this file look very fragile and error-prone.  Wondering if it should be reworked in this PR or a future one.